particle-builtins/src/builtins.rs

```diff
@@ -806,6 +835,40 @@ where
     FunctionOutcome::Ok(json!(out))
 }
 
+#[derive(thiserror::Error, Debug, PartialEq)]
+enum ResolveVaultError {
+    #[error("Incorrect vault path `{0}`: doesn't belong to vault (`{1}`)")]
+    WrongVault(path::PathBuf, path::PathBuf),
+    #[error("Incorrect vault  path `{0}`: doesn't exist")]
+    NotFound(path::PathBuf),
```

Comment:
Suggest to add original error to each of these enum variants.

```suggestion
    #[error("Incorrect vault path `{0}`: doesn't belong to vault (`{1}`)")]
    WrongVault(#[source] StripPrefixError, path::PathBuf, path::PathBuf),
    #[error("Incorrect vault  path `{0}`: doesn't exist")]
    NotFound(#[source] io::Error, path::PathBuf),
```